File: Irrgarten/src/irrgarten/Shield.java
package irrgarten;

/**
 *
 * @author juanma
 */
public class Shield {

    private float protection;
    
    private int uses;
    
    public Shield(float iProtection, int iUses){
        protection = iProtection;
        uses = iUses;
    }
    public float protect(){
        if(uses > 0 ){
            uses--;
            return protection;
        }
        return 0;
    }
    
    public boolean discard(){
        return Dice.discardElement(uses);
    }
    
    public String toString(){
        String _protection = Float.toString(protection);
        String _uses = Integer.toString(uses);
        String cadena  = "S[" + _protection + ", " + _uses + "]";
        return cadena;
    }
}

File: IrrgartenRuby/Shield.rb
module Irrgarten
  class Shield
    def initialize(iProtection, iUses)
      @protection = iProtection
      @uses = iUses
    end
    
    def protect
      if(@uses > 0) then
        @uses -= 1
        return @protection
      end
      0
    end

    def to_s
      "S[#{@protection}, #{@uses}]"
    end

    def discard
      Dice.discardElement(@uses)
    end


  end
  
end
